Library/Homebrew/rubocops/deprecate.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require "rubocops/extend/formula"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      # This cop audits deprecate!
+      class Deprecate < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, body_node)
+          deprecate_node = find_node_method_by_name(body_node, :deprecate!)
+
+          return if deprecate_node.nil? || deprecate_node.children.length < 3
+
+          date_string = find_strings(deprecate_node).first
+
+          return if regex_match_group(date_string, /^\d{4}-\d{2}-\d{2}$/)
+
+          offending_node(date_string)
+          problem "Deprecation date should comply with ISO-8601 (YYYY-MM-DD)"
```

Comment:
Would it be cheeky to include a link to https://xkcd.com/1179/ ?

Reply:
Haha, why not? It might just brighten someone's day...